src/fio/FioEngine.ts

```diff
@@ -1098,18 +1059,76 @@ export class FioEngine extends CurrencyEngine<FioTools, SafeFioWalletInfo> {
       }
 
       if (areDomainsChanged) {
-        isChanged = true
         this.otherData.fioDomains = result.fio_domains.map(fioDomain => ({
           name: fioDomain.fio_domain,
           expiration: fioDomain.expiration,
           isPublic: fioDomain.is_public === 1
         }))
+        return true
       }
     } catch (e: any) {
-      this.warn('checkAccountInnerLoop getFioNames error: ', e)
+      this.warn('fetchDomains getFioDomains error: ', e)
     }
 
-    if (isChanged) this.localDataDirty()
+    return false
+  }
+
+  // Check all account balance and other relevant info
+  async checkAccountInnerLoop(): Promise<void> {
+    const currencyCode = this.currencyInfo.currencyCode
+
+    // Initialize balance
+    if (
+      typeof this.walletLocalData.totalBalances[currencyCode] === 'undefined'
+    ) {
+      this.updateBalance(currencyCode, '0')
+    }
+
+    // Balance
+    try {
+      const balances = { staked: '0', locked: '0' }
+      let balanceRes: BalanceResponse
+      try {
+        balanceRes = asGetFioBalanceResponse(
+          await this.multicastServers('getFioBalance')
+        )
+      } catch (e: any) {
+        if (e?.json?.message === MAINNET_LOCKS_ERROR) {
+          balanceRes = asGetFioBalanceResponse(
+            await this.multicastServers('getCurrencyBalance')
+          )
+          this.log.warn('Returning FIO balance only due to chain bug')
+        } else {
+          throw e
+        }
+      }
+      const { balance, available, staked, srps, roe } = balanceRes
+      const nativeAmount = String(balance)
+      balances.staked = String(staked)
+      balances.locked = sub(nativeAmount, String(available))
+
+      this.otherData.srps = srps
+      this.otherData.stakingRoe = roe
+
+      this.updateBalance(currencyCode, nativeAmount)
+
+      if (
+        balances.staked !== this.otherData.lockedBalances.staked ||
+        balances.locked !== this.otherData.lockedBalances.locked
+      ) {
+        this.otherData.lockedBalances = balances
+        this.localDataDirty()
+        this.updateStakingStatus()
+      }
+    } catch (e: any) {
+      this.log.warn('checkAccountInnerLoop getFioBalance error: ', e)
+    }
+
+    // Fio Addresses and Domains
+    const isAddressChanged = await this.refreshFioAddresses()
+    const isDomainChanged = await this.refreshFioDomains()
```

Comment:
If you still want error handling in the inner loop, you can just put a try / catch around these two guys specifically.